src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -214,35 +212,19 @@ private void createKNNIndexFromTemplate(Model model, KNNCodecUtil.Pair pair, KNN
 
     private void createKNNIndexFromScratch(FieldInfo fieldInfo, KNNCodecUtil.Pair pair, KNNEngine knnEngine, String indexPath)
         throws IOException {
-        Map<String, Object> parameters = new HashMap<>();
         Map<String, String> fieldAttributes = fieldInfo.attributes();
         String parametersString = fieldAttributes.get(KNNConstants.PARAMETERS);
-
-        // parametersString will be null when legacy mapper is used
         if (parametersString == null) {
-            parameters.put(KNNConstants.SPACE_TYPE, fieldAttributes.getOrDefault(KNNConstants.SPACE_TYPE, SpaceType.DEFAULT.getValue()));
-
-            String efConstruction = fieldAttributes.get(KNNConstants.HNSW_ALGO_EF_CONSTRUCTION);
```

Comment:
But the fieldInfo of old segments will not be re-used to create new segments. We will need to use the fieldInfo that is created for new segments. This fieldInfo will be setup using the MethodFieldMapper which will specify the paramsString. So, I dont see an issue with BWC

Reply:
> But the fieldInfo of old segments will not be re-used to create new segments

FieldInfo as per my understanding is at the Lucene index level(aka shard level) and not the lucene segment level. Please validate this, by doing a local test. If already done please let me know the steps, I will check on my side too.

Reply:
This is what I was referring to: https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/index/FieldInfo.java#L24-L29. FieldInfo is per-segment. I can look a little bit more into this.

Reply:
interesting. But can we do 1 validation to be 100% sure. because fieldInfo gets created per segment when we are searching. But during indexing are there checks which may prevent be triggered same field has different values.

Example, we cannot create a field with docvalues as false for 1 segment and for another segment make it true. So similarly can we check, for same field can we add 1 attribute and then next time we add another attribute. 

Please ignore if we have done this check. Just want to ensure that we have handled/thought about all the cases.

Reply:
This test case covers this case so it should be good: https://github.com/opensearch-project/k-NN/blob/main/qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/IndexingIT.java#L58-L76. 

In terms of rejecting merging for dv, I think this is the chain:
1. https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/index/IndexWriter.java#L3178
2. https://github.com/apache/lucene/blob/1edda101a3c02b00c26bd01b1df0f47e35c386b1/lucene/core/src/java/org/apache/lucene/index/FieldInfos.java#L443
3. https://github.com/apache/lucene/blob/1edda101a3c02b00c26bd01b1df0f47e35c386b1/lucene/core/src/java/org/apache/lucene/index/FieldInfos.java#L547

Nothing is checked around attributes

Reply:
@jmazanec15 if nothing is checked around the attributes then in case of segment merges which value is picked for an attribute if both segments have different values for a given attribute key. I am little confused here.

Reply:
I did some testing of my own, like this on local.
1. Added a attribute with key `navneet` and value `200` in the fieldType in code.
2. Then started the cluster and ingested 3 docs and validated using break point in search that 200 value is present.
3. Now I killed the cluster and change the code with value as `300`.
4. Now ran the cluster with `--preserve-data` flag and then ingested 3 more docs in the cluster.
5. ran the query and using breakpoint I can see segment 1 with `200` value and segment 2 with `300` value.
6. Now I force merge the segment to 1. Ran the search again and now I can see only `300` value for that attribute.

**Observations:**
1. If values of attributes is changing then indexing doesn't fail. which is good.
2. When we do the merge only 1 value will be retained and it seems like latest segment value will be there. But I am not sure. This is not an exhaustive test.

Reply:
@jmazanec15 I thought of another use case and want to know how BWC is maintained here. If lets say 2 old segments are merged where parameters were null. What will happen in that case? Because none of the segments had the parameters value it can lead to exception as per my understanding. Whats your thought?

Reply:
> 3. When we do the merge only 1 value will be retained and it seems like latest segment value will be there. But I am not sure. This is not an exhaustive test.

I was thinking that the new fieldType would be inherited from the new FieldMapper. Let me check on this.

> @jmazanec15 I thought of another use case and want to know how BWC is maintained here. If lets say 2 old segments are merged where parameters were null. What will happen in that case? Because none of the segments had the parameters value it can lead to exception as per my understanding. Whats your thought?

Let me check on this

Reply:
@navneet1v 

> @jmazanec15 I thought of another use case and want to know how BWC is maintained here. If lets say 2 old segments are merged where parameters were null. What will happen in that case? Because none of the segments had the parameters value it can lead to exception as per my understanding. Whats your thought?

This case fails: 
```


@jmazanec15 I thought of another use case and want to know how BWC is maintained here. If lets say 2 old segments are merged where parameters were null. What will happen in that case? Because none of the segments had the parameters value it can lead to exception as per my understanding. Whats your thought?
```

That being said, I think we probably need to leave the code as is in the KNN80DocValuesConsumer. 

For field type/mapping, let me check which they use for initializing the merged segments fileinfo. If its always the latest one, then I think we can leave as is.

Reply:
@navneet1v I think this is the [key point](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/index/FieldInfos.java#L732-L734):
```
     * <p>If the field already exists: 1) the provided FieldInfo's schema is checked against the
     * existing field and 2) the provided FieldInfo's attributes are added to the existing
     * FieldInfo's attributes.
```

So, what we can conclude is that the field info attributes will be a Union of the merged fields. So, we just need to keep the legacy checks in KNN80DocValuesConsumer but we can keep Mapper layer as is in this PR.

Reply:
Can we add that failure case in bwc test?

Reply:
Sure @heemin32 added